tests/conftest.py

```diff
@@ -365,21 +365,41 @@ def mk_slash_votes(validator_index, privkey):
 
 
 @pytest.fixture
-def mk_logout():
-    def mk_logout(validator_index, epoch, key):
-        sighash = utils.sha3(rlp.encode([validator_index, epoch]))
-        v, r, s = utils.ecdsa_raw_sign(sighash, key)
+def mk_logout_msg_signed():
+    def mk_logout_msg_signed(validator_index, epoch, key):
+        msg_hash = utils.sha3(rlp.encode([validator_index, epoch]))
+        v, r, s = utils.ecdsa_raw_sign(msg_hash, key)
         sig = utils.encode_int32(v) + utils.encode_int32(r) + utils.encode_int32(s)
         return rlp.encode([validator_index, epoch, sig])
-    return mk_logout
+    return mk_logout_msg_signed
 
 
 @pytest.fixture
-def logout_validator(casper, mk_logout):
-    def logout_validator(validator_index, key):
-        logout_tx = mk_logout(validator_index, casper.current_epoch(), key)
-        casper.logout(logout_tx)
-    return logout_validator
+def mk_logout_msg_unsigned():
+    def mk_logout_msg_unsigned(validator_index, epoch):
+        v, r, s = (0, 0, 0)
+        sig = utils.encode_int32(v) + utils.encode_int32(r) + utils.encode_int32(s)
+        return rlp.encode([validator_index, epoch, sig])
+    return mk_logout_msg_unsigned
+
+
+@pytest.fixture
+def logout_validator_via_signed_msg(casper, mk_logout_msg_signed):
+    def logout_validator_via_signed_msg(validator_index, msg_signing_key,
+                                        tx_sender_key=tester.k9):
```

Comment:
@djrtwo I have used the tester.k9 key here.. I was trying to avoid defaulting it to the msg_signing_key which would mean all previous tests would use the withdraw_addr == msg.sender method of logout auth (instead of validate_signature). This feels a bit yuck to me... maybe we could default it to simply 1? Keen to hear your thoughts!

Reply:
How about `42`?

`tester.k9` isn't terrible, but it'd be nice to know we don't accidentally use k9 in some other test and not test what we actually meant to.

As for the testing setup in general... it's all a bit messy using the old tooling around pyethereum. I'm excited to move to web3.py soon :)